src/jit/lower.cpp

```diff
@@ -628,20 +620,14 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
 
                 // Now, build the conditional statement for the current case that is
                 // being evaluated:
-                // GT_JTRUE
-                //   |__ GT_COND
-                //          |____GT_EQ
-                //                 |____ (switchIndex) (The temp variable)
-                //                 |____ (ICon)        (The actual case constant)
-                GenTreePtr gtCaseCond =
-                    comp->gtNewOperNode(GT_EQ, TYP_INT, comp->gtNewLclvNode(tempLclNum, tempLclType),
-                                        comp->gtNewIconNode(i, TYP_INT));
-                /* Increment the lvRefCnt and lvRefCntWtd for temp */
+                // GT_CMP switchValue, caseValue
+                // GT_JCC(EQ)
```

Comment:
> I'd be interested turning operations of 'or', 'xor' and 'and' of power of 2 (or 'not' pow 2) into their variants of bt{x}

I see couple of potential uses for BTC/R/S:
* Avoid uncontained immediates like I've done for `test`. Should be relatively easy to implement but as seen in the `TEST` to `BT` case there aren't many opportunities for this, 64 bit "flags" aren't that common.
* `x | (1 << y)` (and the similar `&` and `^` variants). I suspect these are rare but this is an interesting experiment. All `BT mem, reg` instructions have poor performance and should be avoided. But how do we know in lowering if the first operand will be in memory or a register? The register allocator may or may not allocate a register to a lclvar.
* Actually use `BT{x}` for what it does best - a bit test and a bit change in the same instruction:
```C#
if ((x & 4) == 0) {
    x |= 4;
    // more code
}
```
Not sure if this can be reasonably implemented without SSA. And interestingly enough out of all popular native compilers ICC appears to be only one that handles this pattern.